edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -125,27 +127,34 @@ impl ModuleRegistry for DockerModuleRuntime {
 
         let response = creds
             .map(|creds| {
-                debug!("Pulling {}", image);
                 self.client
                     .image_api()
                     .image_create(&image, "", "", "", "", &creds, "")
-                    .map_err(|err| {
-                        let e = Error::from_docker_error(
+                    .then(|result| match result {
+                        Ok(()) => Ok(image),
+                        Err(err) => Err(Error::from_docker_error(
                             err,
                             ErrorKind::RegistryOperation(RegistryOperation::PullImage(image)),
-                        );
-                        warn!("Attempt to pull image failed.");
-                        log_failure(Level::Warn, &e);
-                        e
+                        )),
                     })
             }).into_future()
-            .flatten();
+            .flatten()
+            .then(move |result| match result {
+                Ok(image) => {
+                    info!("Successfully pulled image {}", image);
+                    Ok(())
+                }
+                Err(err) => {
+                    log_failure(Level::Warn, &err);
```

Comment:
Does this say that the pull failed?

Reply:
Yes, it comes from the context, which in this case is [`RegistryOperation::PullImage`](https://github.com/Azure/iotedge/blob/master/edgelet/edgelet-core/src/module.rs#L412)